core/common/src/main/java/alluxio/wire/TieredIdentity.java

```diff
@@ -58,9 +64,37 @@ public TieredIdentity(List<LocalityTier> tiers) {
    * @return the corresponding wire type tiered identity
    */
   public static TieredIdentity fromThrift(alluxio.thrift.TieredIdentity tieredIdentity) {
+    if (tieredIdentity == null) {
+      return null;
+    }
     return new TieredIdentity(tieredIdentity.getTiers().stream()
-        .map(LocalityTier::fromThrift).collect(Collectors.toList())
-    );
+        .map(LocalityTier::fromThrift).collect(Collectors.toList()));
+  }
+
+  /**
+   * @param identities the tiered identities to compare to
+   * @return the identity closest to this one
+   */
+  public Optional<TieredIdentity> nearest(List<TieredIdentity> identities) {
+    Preconditions.checkState(!identities.isEmpty(), "No identities given");
+    for (LocalityTier tier : mTiers) {
+      for (TieredIdentity identity : identities) {
+        for (LocalityTier otherTier : identity.mTiers) {
+          if (tier.mTierName.equals(otherTier.mTierName)
+              && tier.mValue != null
+              && tier.mValue.equals(otherTier.mValue)) {
+            return Optional.of(identity);
+          }
+        }
+      }
+      // Negative wait for a tier indicates that we should never return identities that do not match
+      // in that tier.
+      if (Configuration.containsKey(Template.LOCALITY_TIER_WAIT.format(tier.getTierName()))
+          && Configuration.getInt(Template.LOCALITY_TIER_WAIT.format(tier.getTierName())) < 0) {
+        return Optional.empty();
+      }
+    }
+    return Optional.of(identities.get(0));
```

Comment:
Why is this the default return?

Reply:
if we reach here, none of the locations matched in any tiers, so the first location is as close as any other.

if callers want a non-deterministic result in this case, they can shuffle before calling `nearest`, like we do in `LocalFirstPolicy`

Reply:
yeah, the return value is tricky, since `empty` means something different. These intricacies should be mentioned in the comments.